kong/tools/stream_api.lua

```diff
@@ -37,7 +37,8 @@ local MAX_DATA_LEN = 2^22 - 1
 
 local HEADER_LEN = #st_pack(PACK_F, MAX_KEY_LEN, MAX_DATA_LEN)
 
-local SOCKET_PATH = "unix:" .. ngx.config.prefix() .. "/stream_rpc.sock"
+-- this module may be loaded before `kong.configuration` is initialized
+local SOCKET_PATH = "unix:" .. ngx.config.prefix() .. "/sockets/stream_rpc.sock"
```

Comment:
this is where we should move to constants. even we are doing inferring, it can help
us make sure there's only one source of truth.

```lua
local SOCKET_PATH = "unix:" .. ngx.config.prefix() .. "/" .. constants.SOCKET_DIRECTORY .. "/stream_rpc.sock"

conf.socket_path = conf.prefix .. "/" .. constants.SOCKET_DIRECTORY
```

Reply:
`constants.DEFAULT_SOCKET_DIRECTORY` is better?